### ISSUE TITLE
Missing support for PyTorch `torch` members

### ISSUE THREAD

**Issue by kwikwag** [PyCQA/pylint#7937]
### Bug description

```python
"""Test pylint whitelist"""

import torch

print(torch.tensor([1]))
print(torch.this_is_an_error)
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
$ pylint a.py
************* Module a
a.py:5:6: E1101: Module 'torch' has no 'tensor' member; maybe 'Tensor'? (no-member)
a.py:6:6: E1101: Module 'torch' has no 'this_is_an_error' member (no-member)

--------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 10.00/10, -10.00)

$ pylint --generated-members=torch.* a.py

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 0.00/10, +10.00)
```

### Expected behavior

Line 5 should not show an error. Line 6 should show an error.
```
************* Module a
a.py:6:6: E1101: Module 'torch' has no 'this_is_an_error' member (no-member)

--------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 10.00/10, -10.00)
```

Invoking with `--generated-members=torch.*` (see above) as suggested by https://github.com/PyCQA/pylint/issues/2067 produces no error and thus is not a proper solution.

### Pylint version

```shell
pylint 2.15.8
astroid 2.12.13
Python 3.9.15 | packaged by conda-forge | (main, Nov 22 2022, 08:39:05) [MSC v.1929 64 bit (AMD64)]
```

### OS / Environment

Windows 10 64-bit, conda 4.13.0

### Additional dependencies

torch==1.13.0

**Comment by Pierre-Sassoulas** [comment]
There's no way for us to parse torch's code to understand the generated part, in fact the ``--generated-members`` option is there to tell us what is valid or not yourself. When using ``torch.*`` it signal to pylint that everything is okay. So the result is expected. You're going to have to give another regex to pylint something like ``torch.tensor`` or something more complicated than that if there are other generated member.

**Comment by yuval-gvs** [comment]
FWIW I'll mention (a) that I did try `--extension-pkg-allow-list` and it was not working; and that I opened an issue in the PyTorch repo -- https://github.com/pytorch/pytorch/issues/90827. Hope this gets solved one way or another.

**Comment by yuval-gvs** [comment]
FWIW I'll mention (a) that I did try `--extension-pkg-allow-list=torch` and it was not helping; and (b) that I opened an issue in the PyTorch repo -- https://github.com/pytorch/pytorch/issues/90827. Hope this gets solved one way or another.
